packages/artifact/src/internal/upload/upload-zip-specification.ts

```diff
@@ -75,10 +81,11 @@ export function getUploadZipSpecification(
           - file3.txt
   */
   for (let file of filesToZip) {
-    if (!fs.existsSync(file)) {
+    const stats = fs.lstatSync(file, {throwIfNoEntry: false})
+    if (!stats) {
       throw new Error(`File ${file} does not exist`)
     }
-    if (!fs.statSync(file).isDirectory()) {
+    if (!stats.isDirectory()) {
```

Comment:
Can you have a symbolic link to a directory?
Do we need a normal stat to check this?

Reply:
> Can you have a symbolic link to a directory?

Yes you can!

> Do we need a normal stat to check this?

I don't think so. As long as we know it's a directory that's all that matters. We add the symlink's path as the `destinationPath` below will a `sourcePath` of `null`, so when we create the directory in the zip, it's just a normal directory.

Directory symlinks don't need to be explicitly followed since [our globber will do that for us](https://github.com/actions/upload-artifact/blob/3eadd8b791cabf7cd572f194da82158c24125bd8/src/shared/search.ts#L16) when traversing the files. So as long as we know that it's a directory (symlink or not), that's all that matters.